### PR TITLE
feat: ui.inline_alert

### LINKED ISSUE #867: docs: ui.inline_alert
Blocked by implementation.

### LINKED ISSUE #937: ui.inline_alert
Spectrum docs: https://react-spectrum.adobe.com/react-spectrum/InlineAlert.html
Docs ticket: #867

### PR DESCRIPTION
**Pull request by AkshatJawne**
Closes #937 and Closes #867

### CODE REVIEW

**Review by mofojed — DISMISSED** [review]
Code looks good. Should also add to e2e tests, `ui_render_all.py` and update snapshots (we've missed this with a couple of the last few components merged I think).

**Review by margaretkennedy — DISMISSED** [review]

File: plugins/ui/docs/components/inline_alert.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    ui.heading("Payment Information"),
+    ui.content(
+        "Enter your billing address, shipping address, and payment method to complete your purchase."
+    ),
+)
+```
+
+## UI Recommendations
+
+Consider using [`ui.heading`](./heading.md) if you want to create different types of headings.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/inline_alert.md:21]: Without first knowing an inline alert contains a ui.heading, this feels out of context.

File: plugins/ui/docs/components/inline_alert.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    ui.content(
+        "Enter your billing address, shipping address, and payment method to complete your purchase."
+    ),
+)
+```
+
+
+## Variant
+
+A variant can be set using the `variant` prop to give inline alerts a semantic meaning.
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/inline_alert.md:44]:
```suggestion
The `variant` prop can set a variant to give inline alerts a semantic meaning.
```

File: plugins/ui/docs/components/inline_alert.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            variant="positive",
+        ),
+        ui.inline_alert(
+            ui.heading("Payment Information"),
+            ui.content(
+                "Enter your billing address, shipping address, and payment method to complete your purchase."
+            ),
+            variant="notice",
+        ),
+        ui.inline_alert(
```
Comment (margaretkennedy) [inline plugins/ui/docs/components/inline_alert.md:74]: should this be the same as above? not sure what notice vs negative should do here

Comment (AkshatJawne) [reply plugins/ui/docs/components/inline_alert.md:74]: Screenshot 2024-11-15 at 12 47 08 PM
The notice has a yellow outline, making it slightly different.

**Review by mofojed — APPROVED** [review]
Thanks for adding `ui.badge` and `ui.link` as well!

**Review by margaretkennedy — APPROVED** [review]

**Review by dsmmcken — CHANGES_REQUESTED** [review]
Heading is required to be styled correctly, should this have the same structure as contextual help?

**Review by margaretkennedy — APPROVED** [review]

File: plugins/ui/docs/components/inline_alert.md [hunk 4]
```diff
@@ -4,33 +4,15 @@ Inline alerts display non-modal messages related to objects in a view, often use
...
-    ),
-)
-```
-
-
-## Content
-
-The inline alerts contain a title and body using the `ui.heading` and `ui.content`.
-
+For the contextual help component, both the `heading` and `content` props are required.
```
Comment (mofojed) [inline plugins/ui/docs/components/inline_alert.md:7]:
```suggestion
For the inline alert component, both the `heading` and `content` props are required. 
```

**Review by mofojed — CHANGES_REQUESTED** [review]

**Review by dsmmcken — APPROVED** [review]

**Review by margaretkennedy — APPROVED** [review]

**Pull request merged**